### ISSUE TITLE
Simple synchronous route seems to block app

### ISSUE THREAD

**Issue by abidlabs** [tiangolo/fastapi#4453]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Request, Depends, HTTPException, status
import uvicorn
import time

app = FastAPI()

@app.get('/')
def main():
    time.sleep(5)
    return []

uvicorn.run(app, host="127.0.0.1", port=5001)
```

### Description

I expect this code to be non-blocking, since it is declared using a regular `def` instead of `async def`, meaning each time I open the main route, it should load in a separate thread. However, when I run this application, and open up the `/` route on multiple tabs in my browser, it is clear that the app is running synchronously. It takes 5 seconds for each tab to load the route, starting from the time the previous tab is loaded. So if I open 4 tabs, it takes about 20 seconds for the last tab to load.

Am I misunderstanding something? Is there a way to get code like this to run so that each time the page is loaded, it doesn't block other pages from being loaded?

### Operating System

Windows

### FastAPI Version

0.70.1

### Python Version

3.9.5

**Comment by lucastosetto** [comment]
@abidlabs Sorry if I missed something, but wasn't it supposed to be the other way around? Have you tried the same code inside an `async def` function?

**Comment by MatthijsKok** [comment]
I cannot reproduce your problem.  
Using the following code in pycharm, and opening 4 tabs as fast as possible after eachother:  
All the tabs returning after 5 seconds in the same order and speed as I opened them.
```
from fastapi import FastAPI
import uvicorn
import time

app = FastAPI()


@app.get('/')
def main():
    time.sleep(5)
    return []


if __name__ == '__main__':
    uvicorn.run(app, host="127.0.0.1", port=5001)
```

**Comment by abidlabs** [comment]
> @abidlabs Sorry if I missed something, but wasn't it supposed to be the other way around? Have you tried the same code inside an `async def` function?

Nope, as far as I understand, a route in a `def` should create different threads, while an `async def` should not

**Comment by abidlabs** [comment]
> I cannot reproduce your problem. Using the following code in pycharm, and opening 4 tabs as fast as possible after eachother: All the tabs returning after 5 seconds in the same order and speed as I opened them.
> 
> ```
> from fastapi import FastAPI
> import uvicorn
> import time
> 
> app = FastAPI()
> 
> 
> @app.get('/')
> def main():
>     time.sleep(5)
>     return []
> 
> 
> if __name__ == '__main__':
>     uvicorn.run(app, host="127.0.0.1", port=5001)
> ```

Thanks for testing @MatthijsKok. When I use `cURL` to test, I do not see the problem, but I do see it with my browser for some reason. I don't know the reason but I guess it doesn't have to do with FastAPI so I'll close the issue.

**Comment by ycd** [comment]
This is not related to the FastAPI, conversely it is a browser feature. 

See the similar issues:
https://github.com/tiangolo/fastapi/issues/2786
https://github.com/tiangolo/fastapi/issues/3697

In chromium there is a cache that prevents sending the same requests to same URL to avoid downloading the same data over and over again by adding a 20 second timeout to cache. 

If you are interested in how it works check this issue in chromium.

See: https://codereview.chromium.org/345643003

**Comment by abidlabs** [comment]
Ohhhhhh, interesting. Thanks for letting me know!
